frontend_tests/zjsunit/namespace.js

```diff
@@ -22,6 +22,47 @@ const real_jquery_path = require.resolve("../zjsunit/real_jquery.js");
 let in_mid_render = false;
 let jquery_function;
 
+const TEMPLATE_PATH = "/static/templates/";
+
+function need_to_mock_template_error(filename) {
+    const i = filename.indexOf(TEMPLATE_PATH);
+
+    if (i < 0) {
+        throw new Error("programming error");
+    }
+
+    const fn = filename.slice(i + TEMPLATE_PATH.length);
+
+    return `
+        Please use mock_template if your test needs to render ${fn}
+
+        The basic pattern is this (grep for mock_template to see real
+        world examples):
+
+        run_test("test something calling template", ({mock_template}) => {
+            // We encourage you to set the second argument to false
+            // if you are not actually inspecting or using the results
+            // of actually rendering the template.
+            mock_template("${fn}", false, (data) => {
+                assert.deepEqual(data, {...};
+                // or assert more specific things about the data
+                return "stub-for-zjquery";
+            });
+
+            // If you need the actual HTML from the template, do
+            // something like below instead. (We set the second argument
+            // to true which tells mock_template that is should call
+            // the actual template rendering function and pass in the
+            // resulting html to us.
```

Comment:
I think these comments might benefit from setting a bit more context for someone who is new to Zulip.  Something like (not complete, just trying to give the idea):

```
This test runs JavaScript code that renders the {fn} HTML template, but does not currently test that step. You can test template rendering using the mock_template method in two ways. In both cases, you call `mock_template` in the setup code for the test before calling whichever function actually renders the template:

run_test("test template by inspecting rendering parameters", ({mock_template}) => {
// If you don't need the HTML generated by rendering the template, this method lets you 
// verify the context passed into the template rendering function:
mock_template("${fn}", false, (data) => {
   assert.deepEqual(data, {...};
   // What you return will be treated as the HTML generated by rendering the template.
   return "stub-for-zjquery";
}

mock_template("${fn}", true, (data, html) => {
   // comment here
   assert.deepEqual(data, {...};
   assert.ok(html.startWith(...));
   return html;
}
```

Reply:
Comment added.